themes/classic/templates/checkout/_partials/cart-voucher.tpl

```diff
@@ -3,7 +3,8 @@
     <div class="cart-voucher">
       <ul class="promo-name">
         {foreach from=$cart.vouchers.added item=voucher}
-          <li>{$voucher.name} <a class="pull-xs-right" href="{$voucher.delete_url}" data-link-action="remove-voucher">{l s='Cancel'}</a></li>
+          <li>{$voucher.name} {if $voucher.reduction_amount != '0.00'}{$voucher.reduction_amount_formatted} {/if} {if $voucher.reduction_percent != '0.00'} {$voucher.reduction_percent}%{/if}
```

Comment:
The .tpl file does not have to calculate if it's a percentage or an amount, the presenter should give you the formatted string. Could you change that, please ?
Also, of the value is 0, it should not be displayed, so the presenter should not send the voucher to the .tpl